controllers/component_controller.go

```diff
@@ -172,7 +172,8 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	log.Info(fmt.Sprintf("Starting reconcile loop for %v", req.NamespacedName))
 
 	if component.Spec.ContainerImage == "" {
-		component.Spec.ContainerImage = r.ImageRepository + ":" + component.Namespace + "-" + component.Name
+		uniqueHash := util.GenerateUniqueHashForWorkloadImageTag(req.ClusterName, component.Namespace)
```

Comment:
It comes from pr 203 which is also mentioned in PLNSRVCE-773. When the jira issue was opened and pr 203 was opened, kcp was being used. After moving back to OSD cluster, changes in pr 203 hasn't been dropped yet. I've been being confused and asked whether PLNSRVCE-773 is still a valid request.

Back to the time when using kcp, `req.ClusterName` has value related to kcp enviornment and the solution introduced in pr 203 worked to ensure the uniqueness by combining `req.ClusterName` and namespace. But, nowadays, without kcp, I think the solution of DEVHAS-176 and PLNSRVCE-773 has to be reconsidered. Probably it would be good to open a new jira story for that.

@Michkov What do you think?

Reply:
let's just re-use/refactor current code for reaching goal of avoiding to expose username. Refactoring due to kcp changes is out-of-scope for this change.

Reply:
@Michkov What do you mean by "current code"?

Reply:
basically I agree with your current solution. Let's not introduce additional cleanups.